agent/agent.go

```diff
@@ -5,13 +5,15 @@ import (
 
 	"github.com/negasus/haproxy-spoe-go/logger"
 	"github.com/negasus/haproxy-spoe-go/request"
+	"github.com/negasus/haproxy-spoe-go/stats"
 	"github.com/negasus/haproxy-spoe-go/worker"
 )
 
-func New(handler func(*request.Request), logger logger.Logger) *Agent {
+func New(handler func(*request.Request), logger logger.Logger, statter stats.Statter) *Agent {
```

Comment:
Note to self. Make this a `WithStatter` func that wraps the agent so we don't need to change this signature